### ISSUE TITLE
missing search parameter in HyperOptSearch

### ISSUE THREAD

**Issue by wxie2013** [ray-project/ray#49507]
### What happened + What you expected to happen

I have a set of pre-tuned hyperparameters. Now I want to tune other parameters while randomly sampling the pre-tuned parameters.  When using HyperOptSearch, those pre-tuned parameters are somehow lost during the new tuning process. If I remove "search_algo = HyperOptSearch(...)', the pre-tuned parameters do show up. 

### Versions / Dependencies

ray 2.40. 0

### Reproduction script

      import ray
      from ray import tune
      import numpy as np
      from ray.tune.search.hyperopt import HyperOptSearch

      # Define the existing hyperparameters as a list of dictionaries
      existing_hyperparameters = [
          {"learning_rate": 0.01, "batch_size": 32},
          {"learning_rate": 0.1, "batch_size": 2},
      ]

      # Define the new extra hyperparameter to optimize
      new_hyperparameter = {"new_param": tune.uniform(0.1, 10)}

      # Define the search space
      search_space = {
          "existing_hyperparameters": tune.choice(existing_hyperparameters),
          **new_hyperparameter,
      }

      # Define the objective function to optimize
      def objective(config):
          existing_hyperparameters = config["existing_hyperparameters"]
          new_param = config["new_param"]
          # Simulate some objective function evaluation
          return {"result": 0}

      # Initialize Ray
      ray.init()

      # Create a Tuner object
      tuner = tune.Tuner(
          objective,
          param_space=search_space,
          tune_config=tune.TuneConfig(
              num_samples=1,
              #search_alg=HyperOptSearch( metric="result", mode="max"),
              )
      )

      # Run the hyperparameter search
      results = tuner.fit()
      # Shutdown Ray

      ray.shutdown()

### Issue Severity

High: It blocks me from completing my task.
